src/sage/combinat/permutation.py

```diff
@@ -7092,7 +7092,17 @@ def _element_constructor_(self, x, check=True):
             [1, 4, 5, 2, 3, 6]
             sage: Permutations(6)(x)            # known bug
             [1, 4, 5, 2, 3, 6]
+
+        Ensure that :issue:`37284` is fixed::
+
+            sage: S5 = SymmetricGroup(5)
+            sage: P5 = Permutations(5)
+            sage: p = S5.list()[3]
+            sage: P5(p)
+            [4, 5, 1, 2, 3]
         """
+        if isinstance(x, PermutationGroupElement):
+            x = x.domain()
```

Comment:
```suggestion
            return self. _from_permutation_group_element(x)
```
No, you should only make this change here. Do not change any coercions!